lib/src/apigateway/openapi-gateway-lambda.ts

```diff
@@ -115,14 +115,34 @@ const validateOpenapiOperations = (operations: LambdaOperation[]): void => {
 const addGatewayToLambdaPermissions = (
   scope: Construct,
   lambdaOperations: LambdaOperation[],
+  specRestApi: SpecRestApi,
 ): void => {
-  if (!lambdaOperations || lambdaOperations.length === 0) {
-    throw new Error('lambdaOperations prop is required and should be non-empty');
-  }
+  const addedPermissionLambdaSet = new Set();
+
   lambdaOperations.forEach((lambdaOperation: LambdaOperation) => {
-    lambdaOperation.lambdaAlias.addPermission('allow-apigw-to-lambda', {
-      principal: new ServicePrincipal('apigateway.amazonaws.com'),
-    });
+    // ? this `functionArn` is the alias arn
```

Comment:
I never did it. Maybe it's difficult because we are talking about getting the Arn (which is created only during CFN deployment) of a resource two levels up and maybe we need to do some tricks to do that.

For now keeping in the alias is good enough. Actually maybe that's why my initial implementation was that way too :)